### DISCUSSION TITLE
stabilityDays taking up prConcurrentLimit slots

### DISCUSSION THREAD

**Discussion by dbasilio** [renovatebot/renovate#15105] [Request Help] [answered]
Our config has the following options set:
```json
{
    "prCreation": "not-pending",
    "stabilityDays": 21,
    "prConcurrentLimit": 2,
}
```

The behaviour we're seeing is that PRs that have not met stability days take up one of the `prConcurrentLimit` slots. For example, if we have the following package rules:
```json
{
    "packageRules": [
         {
             "matchPackagePrefixes": ["packageA"],
             "prPriority": 1
         },
         {
             "matchPackagePrefixes": ["packageB"],
             "prPriority": 2
         },
         {
             "matchPackagePrefixes": ["packageC"],
             "prPriority": 3
         },
    ]
}
```
All 3 packages have an update available, package C and A have already passed their stability days requirement, but package B has not. I would expect that we would get PRs for package A and C, and once the stability days for package B were met, we would get a PR for it. Instead, what we're seeing via the dependency dashboard is the following:
- Run 1:
  - package C gets a branch
  - package B gets a branch
- Run 2:
  - Package C passed all checks, so it gets a PR
  - Package B fails stability days check, no change
- Run 3:
  - Package C no change
  - Package B fails stability days check, no change

Package B eats up one of the slots of our `prConcurrentLimit`, **even though it doesn't have a PR**, meanwhile package A could have a PR put up but is ignored. I would have expected package A to get a branch on Run 2 and a PR on Run 3 (or Runs 3 and 4 respectively).

Since we use a fairly length stability days, that means we sometimes end up with all of our PR slots taken up by packages we can't merge for several weeks, meanwhile lower priority packages that we could be merging are not considered at all.

I'm not sure if this is a bug, a feature request, or if we're doing something wrong with our config here.

**Comment by rarkins** [comment, marked as answer]
Look into either branchConcurrentLimit (set it higher than prConcurrentLimit) or internalChecksFilter=strict

**Reply by dbasilio** [reply]
I think `internalChecksFilter=strict` is what we're looking for thank you 🙏
